src/sinks/statsd.rs

```diff
@@ -215,8 +215,12 @@ impl Encoder<Event> for StatsdEncoder {
                     StatisticKind::Histogram => "h",
                     StatisticKind::Summary => "d",
                 };
-                let samples = compress_distribution(samples.clone());
-                for sample in samples {
+
+                // TODO: This is a good example of where it'd be nice to bake this into a dedicated combinator or
+                // normalizer feature, although this sink also needs to be rewritten in the new-style to begin with.
```

Comment:
I think this TODO could use some more elaboration in-code. If I put myself in the seat of someone converting the sink to new-style I don't think what's being discussed here is obvious.

Reply:
Agreed.  I'll try and flesh it out a little bit more.

Ironically, new-style sinks don't actually do this (compress distributions to save as much space when encoded as possible) so there's a tension of coming up with a way to do it for new-style sinks... and also rewrite the `statsd` sink in the new style.

Reply:
Addressed, I hope, via 0945e10.

Reply:
Perfect.